t/t1040-issue478.t

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+test_description='test calling view-user with the --list-banks optional argument'
+
+. `dirname $0`/sharness.sh
+
+mkdir -p conf.d
+
+MULTI_FACTOR_PRIORITY=${FLUX_BUILD_DIR}/src/plugins/.libs/mf_priority.so
+SUBMIT_AS=${SHARNESS_TEST_SRCDIR}/scripts/submit_as.py
+ACCOUNTING_DB=$(pwd)/FluxAccountingTest.db
+
+export TEST_UNDER_FLUX_SCHED_SIMPLE_MODE="limited=1"
+test_under_flux 1 job -o,--config-path=$(pwd)/conf.d
+
+flux setattr log-stderr-level 1
+
+
+test_expect_success 'create flux-accounting DB, start flux-accounting service' '
+	flux account -p ${ACCOUNTING_DB} create-db &&
+	flux account-service -p ${ACCOUNTING_DB} -t
+'
+
+test_expect_success 'add some banks' '
+    flux account add-bank root 1 &&
+    flux account add-bank --parent-bank=root bankA 1 &&
+    flux account add-bank --parent-bank=root bankB 1 &&
+    flux account add-bank --parent-bank=root bankC 1
+'
+
+test_expect_success 'add a user' '
+    flux account add-user --username=testuser --bank=bankA &&
+    flux account add-user --username=testuser --bank=bankB &&
+    flux account add-user --username=testuser --bank=bankC
+'
+
+test_expect_success 'call view-banks --list-banks' '
+	flux account view-user testuser --list-banks > list_banks.out &&
+    grep -E "bankA|bankB|bankC" list_banks.out
```

Comment:
Sorry that I didn't catch this earlier! I usually double check this but it slipped through.

I just force-pushed up a change to also just separate the `grep` check into three different checks for all three banks instead of combining all of them on just one line (that was my misunderstanding of the `-E` option, sorry!). Can you let me know if you are still seeing that weird output behavior when calling `view-user --list-banks`? I wasn't able to reproduce that locally (yet) and I tried calling `view-user` with every optional argument...